### ISSUE TITLE
ESLint is not reporting issues in VS Code; does on command line.

### ISSUE THREAD

**Issue by ruffin--** [microsoft/vscode-eslint#1472]
I have a javascript file `C:\temp\temp.js`.

Its contents are:

```
hello = 'hello';
```

Though the eslint extension is running...

![screenshot showing the output from the extension](https://user-images.githubusercontent.com/5659878/170522280-6b6ee58b-c862-4e66-bce5-347292799455.png)

```
[Info  - 11:23:47 AM] ESLint server is starting
[Info  - 11:23:49 AM] ESLint server running in node v16.13.0
[Info  - 11:23:49 AM] ESLint server is running.
```

... I'm not getting any errors reported.

If I run eslint on the command line using VSCode's terminal, it does return the expected issue.

```
PS C:\temp> eslint temp.js

C:\temp\temp.js
  1:1  error  'hello' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

This is a new behavior. It was working maybe as recently as earlier this week.

Major changes on my system this week?

* Installed node-v16.15.0-x64

This is ESLint extension v2.2.2. 

--------------------------

### VS Code info:

```
Version: 1.67.2 (user setup)
Commit: c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5
Date: 2022-05-17T18:15:52.058Z
Electron: 17.4.1
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.19044
```

### Screenshot of the deets

![simplest case VSCode vs command line with js file, eslint extension on, full eslintrc visible](https://user-images.githubusercontent.com/5659878/170521936-a9a38c39-9cde-4c46-9626-b165c9ad27f7.png)

### c:\temp\eslintrc in full as text

```
{
    "env": {
        "browser": true,
        "es6": true,
        "jasmine": true,
        "jquery": true,
        "node": false
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 6
    },
    "rules": {
        "indent": [
            2,
            4,
            {
                "SwitchCase": 1,
                "ignoredNodes": ["ConditionalExpression"]
            }
        ],
        "no-undef": "error",
        "no-unused-expressions": 2,
        "no-unused-vars": [
            2,
            {
                "args": "none"
            }
        ],
        "quotes": [
            2,
            "single",
            {
                "allowTemplateLiterals": true,
                "avoidEscape": true
            }
        ],
        "semi": [2, "always"]
    }
}
```

**Comment by ruffin--** [comment]
### TL;DR

I didn't run `npm update -g`  after upgrading node?

--------

Adding that to settings actually wasn't the (only?) issue. It worked briefly, but then stopped again after a reboot. 😕 

Based on [this](https://stackoverflow.com/a/68414182/1028230), I added the following to my settings.json for VS Code:

```
  "eslint.validate": [
    "javascript",
    "javascriptreact",
    "typescript",
    "typescriptreact"
  ],
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
  },
```

Now it's running, but saying it can't find eslint.

```
[Info  - 1:27:39 PM] ESLint server is starting
[Info  - 1:27:40 PM] ESLint server running in node v16.13.0
[Info  - 1:27:40 PM] ESLint server is running.
[Info  - 1:27:41 PM] Failed to load the ESLint library for the document c:\temp\temp.js
To use ESLint for single JavaScript file install eslint globally using 'npm install -g eslint'.
If you are using yarn or pnpm instead of npm set the setting `eslint.packageManager` to either `yarn` or `pnpm`
You need to reopen VS Code after installing eslint.
```

If I run it from the command line, eslint (installed globally) still works fine.

```
PS C:\temp> eslint temp.js

C:\temp\temp.js
  1:1  error  'hello' is not defined  no-undef
  2:1  error  'spam' is not defined   no-undef

✖ 2 problems (2 errors, 0 warnings)

PS C:\temp> 
```
----

I tried to put 2 and 2 together (I'd updated Node + eslint said it couldn't find something I'd installed via `npm` before the update) and did an `npm update -g` to see if that shook `eslint` loose. **It's now working.**

Why it'd work on the terminal but not inside VS Code, I can't say I understand. I even tried setting [`eslint.runtime` to `"node"`](https://github.com/Microsoft/vscode-eslint) to use my node, not VS Code's.

`[Info  - h:mm:ss tt] ESLint server running in node vx.y.z` updated to the local version after changing that, but it still didn't find it.

2¢, ymmv, etc.
